### PR TITLE
Net 224: gap before hotblocks

### PR DESCRIPTION
**Pull request by toschoosqd**
### Summary

  Handle archival-to-hotblocks retention gaps as temporary no-data responses.

  When /stream routes a request to hotblocks because the requested first block is above the archival height, the portal now
  checks hotblocks status first. If the request falls before the oldest retained hotblock, it returns a delayed 204 No Content
  instead of forwarding the hotblocks error.

  ### Details

  - Added hotblocks gap detection using StatusData.first_block.
  - Preserved existing behavior when hotblocks status is unavailable or has no data.
  - Reused delayed 204 response construction for both:
      - no hotblocks source above archival height
      - confirmed archival/hotblocks gap
  - Added focused unit tests for the gap predicate and boundary cases.

  ### Validation

  - cargo test hotblocks_gap

### CODE REVIEW

File: src/http_server.rs [hunk 1]
```diff
@@ -428,8 +428,19 @@ async fn run_stream_internal(
         }
 
         // Then try hotblocks storage
-        // TODO: if the query is below the first hotblock, return 500 instead of 400
         _ if dataset.hotblocks.is_some() => {
+            if should_treat_as_hotblocks_gap(
```
Comment (kalabukdima) [inline src/http_server.rs]: Does it always add another request-response before sending the stream request now? It will inevitably add latency to all stream requests, and we're already exceeding the SLO. It's not worth it for this rare situation.

I can see the following options:
- only send an extra request if the response to `/stream` was unsuccessful, or
- pre-fetch hotblocks status and update it in the background occasionally

Comment (toschoosqd) [reply src/http_server.rs]: First option was implemented.
